apisix/discovery/kubernetes/init.lua

```diff
@@ -29,7 +29,10 @@ local core = require("apisix.core")
 local util = require("apisix.cli.util")
 local local_conf = require("apisix.core.config_local").local_conf()
 local informer_factory = require("apisix.discovery.kubernetes.informer_factory")
-local endpoint_dict = ngx.shared.discovery
+local endpoint_dict = ngx.shared["discovery"]
+if not endpoint_dict then
+    error("failed to get ngx.shared.dict discovery when load kubernetes discovery plugin ")
```

Comment:
not clear

Reply:
Sorry,  I didn't get your point.
Your means is error mesage not clear ?

Reply:
Now, I Had move it  into function init_worker()